syncer/syncer.go

```diff
@@ -856,7 +862,16 @@ func (s *Syncer) flushCheckPoints() error {
 		shardMetaSQLs, shardMetaArgs = s.sgk.PrepareFlushSQLs(exceptTableIDs)
 		s.tctx.L().Info("prepare flush sqls", zap.Strings("shard meta sqls", shardMetaSQLs), zap.Reflect("shard meta arguments", shardMetaArgs))
 	}
-	err := s.checkpoint.FlushPointsExcept(exceptTables, shardMetaSQLs, shardMetaArgs)
+
+	if utils.IsContextCanceledError(tctx.Ctx.Err()) {
+		// when canceling (stop-task/pause-task), we still need to flush the checkpoint.
+		var cancel context.CancelFunc
+		tctx, cancel = tcontext.Background().WithTimeout(defaultDBContextTimeout)
+		defer cancel()
+		s.tctx.L().Info("flushing checkpoint when canceling the task", zap.Stringer("checkpoint", s.checkpoint))
+	}
+
+	err := s.checkpoint.FlushPointsExcept(tctx, exceptTables, shardMetaSQLs, shardMetaArgs)
```

Comment:
I'm not sure whether needing different timeout values (rather than the default one used when canceling) here.

Do you prefer to remove the `tctx` argument here?

Reply:
yes, how about @GregoryIan